core/src/views/dropdown.rs

```diff
@@ -3,11 +3,67 @@ use crate::{
     View,
 };
 
+/// The dropdown icon of a dropdown view.
 pub const ICON_DOWN_OPEN: &str = "\u{e75c}";
 
-pub struct Dropdown {}
+/// A dropdown used to display some state with the ability to open a popup with options to change that state.
+///
+/// Usually a dropdown is used in the context of a "combo box" or "list picker" to allow the user to select
+/// from one of several discrete options. The dropdown takes two closures, one which shows the current state
+/// regardless of whether the dropdown is open or closed, and one which shows the contents while it is open.
+///
+/// ## Basic dropdown
+///
+/// A basic dropdown displaying five options that the user can choose from.
+///
+/// ```
+/// # use vizia_core::*;
+/// #
+/// # #[derive(Lens)]
+/// # struct AppData {
+/// #     value: u8,
+/// # }
+/// #
+/// # impl Model for AppData {}
+/// #
+/// # enum AppEvent {
+/// #     SetValue(u8),
+/// # }
+/// #
+/// # let cx = &mut Context::new();
+/// #
+/// # AppData { value: 0 }.build(cx);
+/// #
+/// Dropdown::new(
+///     cx,
+///     |cx| Label::new(cx, AppData::value),
+///     |cx| {
+///         for i in 0..5 {
+///             Label::new(cx, i)
+///                 .on_press(move |cx| {
+///                     cx.emit(AppEvent::SetValue(i));
+///                     cx.emit(PopupEvent::Close); // close the popup
```

Comment:
Could be worth mentioning here that closing the popup when selecting is optional. This allows for something like a dropdown where multiple things can be selected.